p2p/test/transport/transport_test.go

```diff
@@ -798,3 +867,29 @@ func TestConnClosedWhenRemoteCloses(t *testing.T) {
 		})
 	}
 }
+
+func TestConnMatchingAddress(t *testing.T) {
```

Comment:
Here's the new test with the behavior that IIUC is that's expected and was driving most of the changes (although the connection gater ones might be different)